benchmark.go

```diff
@@ -123,7 +140,7 @@ func runWorker(t transport.Transport, m benchmarkMethod, s *benchmarkState, run
 	}
 }
 
-func runBenchmark(out output, logger *zap.Logger, allOpts Options, resolved resolvedProtocolEncoding, m benchmarkMethod) {
+func runBenchmark(out output, logger *zap.Logger, allOpts Options, resolved resolvedProtocolEncoding, methodName string, m benchmarker) {
```

Comment:
I remember in another comment (my bad) I did to do this `methodName` things here. But I actually think we can do better here.

Since benchmarker has a request, we should be able to get the request from the benchmarker. What about having this interface for benchmarker:

```
type benchmarker interface {
	// WarmTransports creates n transports and warms it up by dispatching specified
	// number of warm-up requests.
	WarmTransports(n int, tOpts TransportOptions, resolved resolvedProtocolEncoding, warmupRequests int) ([]peerTransport, error)

	// Call dispatches a request on the provided transport.
	Call(transport.Transport) (time.Duration, error)
	
	Request() *transport.Request
}
```

note: your original proposal was with a method `Method()`.

Reply:
IMO we can keep pass the method name directly in the argument, exposing transport.Request just for extracting method name seems like an overkill to me